src/renderer/screens/manager/Dashboard.js

```diff
@@ -33,9 +37,14 @@ const Dashboard = ({ device, deviceInfo, result, onReset }: Props) => {
   }, [onReset, firmwareUpdateOpened, currentDevice]);
 
   const exec = useCallback(
-    (appOp, targetId, app) =>
-      command("appOpExec")({ appOp, targetId, app, devicePath: device.path }),
-    [device],
+    (appOp, targetId, app) => {
+      if (!hasInstalledApps && appOp.type === "install") {
+        dispatch(setHasInstalledApps(true));
+      }
+
+      return command("appOpExec")({ appOp, targetId, app, devicePath: device.path });
+    },
+    [device, dispatch, hasInstalledApps],
```

Comment:
I think we better not have the `exec` function getting recalculated each time hasInstalledApps changes. I think the `useAppsRunner`could badly glitch if this happen (probably in context of more than one install).

Also more generally I think it could be dangerous to react on an install action. I would instead react to seeing a apps instead `result.installed` (useEffect)

essentially, anytime a user plug a device with apps (even if it's another device) we unlock the situation

Reply:
I initially had it as a `useEffect` to that exactly and couldn't get the setting to be stored for some reason so I resorted to this. If you think it should work like that I'll try to figure out why it wasn't doing it.